### ISSUE TITLE
Recruiter_id - other content?

### ISSUE THREAD

**Issue by Iljadecoster** [CiviCooP/be.aivl.streetimport#50]
This is a conceptual question, after excahnging with the streetrecruitement agency.

For now the Recruiter_id is basically the civi_contact_id of a recruiter. The idea was that we create this for each new recruiter, so the agency informs us about new recruiters and then adds our contact_id in their system to ad it inthe export they generate for us.

But this would slow down the process, as recruiters might change regularly.

The suggestion is to use as a Recruiter_id not a contact_id from civi but the id of the recruiter in the crm of the recruiting agency. And add this id into the civi record of the recruiter to match against that.

Basically quite similar as the donor_id works match welcomecalls with previously imported new streetrecruitments.

Would that be doable? Within acceptable coding time?

**Comment by ErikHommel** [comment]
@Iljadecoster it is doable and I estimate 2-4 hours (development, test and issue fixing). Not too bad but let's wait on the budget discussion?

**Comment by Iljadecoster** [comment]
If it is only 2-4 hour, the just do it. Don't wait for the budget discussion.This is what we need now, so we do haven some budget margins.Ilja--Ilja De CosterAmnesty International Vlaanderen

---

Ilja De Coster
Medewerker Fondsenwerving en Relatiebeheer

Amnesty International Vlaanderen vzw
  
Kerkstraat 156 - B-2060 Antwerpen  
   
T +32 3 270 34 30
M +32 4 96 17 16 19
F +32 3 235 78 12
W www.aivl.be

Volg ons:  
   
 

This email and any files
transmitted with it are confidential and intended solely for the use of
the individual or entity to whom they are addressed. If you have received
this email in error please notify the system manager.
Erik Hommel --- Re: [be.aivl.streetimport] Recruiter_id - other content? (#50) --- Van:"Erik Hommel" notifications@github.comAan:"CiviCooP/be.aivl.streetimport" be.aivl.streetimport@noreply.github.comCc:"Iljadecoster" iljad@aivl.beDatum:di, 7-jul.-2015 18:54Onderwerp:Re: [be.aivl.streetimport] Recruiter_id - other content? (#50)@Iljadecoster it is doable and I estimate 2-4 hours (development, test and issue fixing). Not too bad but let's wait on the budget discussion?

**Comment by ErikHommel** [comment]
Fine, but only if @systopia can finish this before the end of the week so you can also test it before you go on holiday.

**Comment by Iljadecoster** [comment]
Ok ;-)--Ilja De CosterAmnesty International Vlaanderen

---

Ilja De Coster
Medewerker Fondsenwerving en Relatiebeheer

Amnesty International Vlaanderen vzw
  
Kerkstraat 156 - B-2060 Antwerpen  
   
T +32 3 270 34 30
M +32 4 96 17 16 19
F +32 3 235 78 12
W www.aivl.be

Volg ons:  
   
 

This email and any files
transmitted with it are confidential and intended solely for the use of
the individual or entity to whom they are addressed. If you have received
this email in error please notify the system manager.
Erik Hommel --- Re: [be.aivl.streetimport] Recruiter_id - other content? (#50) --- Van:"Erik Hommel" notifications@github.comAan:"CiviCooP/be.aivl.streetimport" be.aivl.streetimport@noreply.github.comCc:"Iljadecoster" iljad@aivl.beDatum:di, 7-jul.-2015 19:30Onderwerp:Re: [be.aivl.streetimport] Recruiter_id - other content? (#50)Fine, but only if @systopia can finish this before the end of the week so you can also test it before you go on holiday.

**Comment by bjendres** [comment]
@ErikHommel: How were you planning on doing this? Can we use the "Externe Donor Identificatie"? Or `external_identifier`? Or a new custom field?

**Comment by Iljadecoster** [comment]
I would use external_identifier - it has now an other use in the amnesty civ.

**Comment by ErikHommel** [comment]
Never! Do not use core fields for other stuff, it will hurt you at some point :-)))
Seriously, I would add a custom field.

Verstuurd vanaf mijn iPad

Op 8 jul. 2015 om 19:50 heeft Iljadecoster <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:

I would use external_identifier - it has now an other use in the amnesty civ.

**Comment by ErikHommel** [comment]
Externe Donor Identificatie Could be used, but I reckon it might be a bit confusing and hackish, using one field for two different things? Add a custom group to contact subtype Werver is best IMO?

Verstuurd vanaf mijn iPad

Op 8 jul. 2015 om 18:24 heeft SYSTOPIA <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:

@ErikHommelhttps://github.com/ErikHommel: How were you planning on doing this? Can we use the "Externe Donor Identificatie"? Or external_identifier? Or a new custom field?

**Comment by bjendres** [comment]
I implemented a solution that creates a new custom group called "Werver Informatie" for contacts of subtype "Werver", along with a custom field called "Extern Werver ID".

This is now used to identify the recruiters for each line. If no such recruiter is found, a new one is created - just like it used to be before.

The changes are still in `dev_50` branch, I'll merge that once @Iljadecoster  hast tested the modification.

**Comment by bjendres** [comment]
merged to `master`
